File: app/src/main/java/com/coolweather/coolweatherjetpack/data/network/BaseRequestListener.java
package com.coolweather.coolweatherjetpack.data.network;


import com.coolweather.coolweatherjetpack.util.toast.ToastUtils;

public abstract class BaseRequestListener<T> {

    public abstract void onSuccess(T response);

    public void onFail(String message) {
        ToastUtils.show(message);
    }

}

File: app/src/main/java/com/coolweather/coolweatherjetpack/data/network/BaseRequestListener.kt
package com.coolweather.coolweatherjetpack.data.network

import com.coolweather.coolweatherjetpack.util.toast.ToastUtils

abstract class BaseRequestListener<T> {

    abstract fun onSuccess(response: T)

    fun onFail(message: String?) {
        ToastUtils.show(message)
    }
}